### PR TITLE
SCF - Assignment 2 Q1

### PR DESCRIPTION
**Pull request by meta-chinmay-jain**
Here is the submission of Assignment 2 Question 1 of Software Construction Fundamentals (SCF)

### CODE REVIEW

File: Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/HexCalc.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
+package com.arithmetic;
+
+/**
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/HexCalc.java:3]: Done

File: Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/HexCalc.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+ * Creates a command-line User Interface.
+ * 
+ * @author Chinmay Jain
+ * @version 1.0
+ * @since 2021-02-26
+ */
+public class HexCalc {
+	private static Integer n1;
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/HexCalc.java:11]: Done

File: Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/Menu.java [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+			System.out.println("\nInvalid input(s)!");
+		}
+	}
+
+	/**
+	 * To convert a number from Hexadecimal to Decimal.
+	 */
+	private static void convertHexToDecimal() {
+		System.out.print("\nEnter Hexadecimal Number: ");
+		num1 = input.nextLine().toUpperCase();
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/Menu.java:208]: Done

File: Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/HexCalc.java [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+	 * @param num  Input number string.
+	 * @param base Base of input string.
+	 * @return Integer Decimal conversion of number string.
+	 */
+	public static int stringToDecimalNumber(String num, int base) {
+		int length = num.length();
+		int power = 1;
+
+		int result = 0;
+		int count;
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/HexCalc.java:154]: moved inside for loop

File: Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/Menu.java [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+	 * @param num This is the input number in String format.
+	 * @return boolean True or False depending on the input string.
+	 */
+	private static boolean checkFormat(String num) {
+		if (num.isEmpty()) {
+			return false;
+		}
+
+		int length = num.length();
+		char numStart;
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/Menu.java:277]: Done

File: Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/Menu.java [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+			System.out.println("\nInvalid input!");
+		}
+	}
+
+	/**
+	 * Gets the number inputs from user.
+	 */
+	private static void getOperands() {
+		System.out.print("\nEnter First Number: ");
+		num1 = input.nextLine().toUpperCase();
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/Menu.java:259]: Done

File: Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/Menu.java [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+			case 10:
+				changeBase();
+				break;
+			case 11:
+				System.exit(0);
+				break;
+			default:
+				break;
+			}
+		} while (choice != 11);
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/Menu.java:83]: Done, remove 'do-while', used 'while' instead

File: Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/Menu.java [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+	}
+
+	/**
+	 * Displays the list of options on the command line.
+	 */
+	private static void getMenuItems() {
+		System.out.println("\nMENU\n");
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/Menu.java:91]: Done

File: Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/Menu.java [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+			System.out.println("\nInvalid input(s)!");
+		}
+	}
+
+	/**
+	 * Gets multiplication of two numbers.
+	 */
+	private static void getMultiplication() {
+		if (checkFormat(num1) && checkFormat(num2)) {
+			result = hexCalc.multiply(num1, num2);
```
[the remark this answers is not in the document]
Comment (meta-chinmay-jain) [reply Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/Menu.java:138]: Done

File: Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/HexCalc.java [hunk 10]
```diff
@@ -213,6 +219,25 @@ public static char toCharVal(int n) {
...
+	 * Checks if the result of multiplication will overflow integer range.
+	 * 
+	 * @param a Input number 1.
+	 * @param b Input number 2.
+	 * @return boolean True if it will overflow, False otherwise.
+	 */
+	private static boolean isMultiplicationOverflow(int a, int b) {
+		// Check if either of them is zero
+		if (a == 0 || b == 0)
+			return false;
```
Comment (meta-love-gupta) [inline Software Construction Fundamentals - GET/Assignment 2/Hex Arithmetic/src/com/arithmetic/HexCalc.java:232]: usage of code without brace

**Review by meta-love-gupta — APPROVED** [review]

**Pull request merged**